src/cmap/auth/mongodb_oidc/token_entry_cache.ts

```diff
@@ -105,6 +132,35 @@ function expirationTime(expiresInSeconds?: number): number {
 /**
  * Create a cache key from the address and username.
  */
-function cacheKey(address: string, username: string): string {
-  return `${address}-${username}`;
+function cacheKey(
+  address: string,
+  username: string,
+  requestFn: OIDCRequestFunction | null,
+  refreshFn: OIDCRefreshFunction | null
+): string {
+  return `${address}-${username}-${hashFunctions(requestFn, refreshFn)}`;
+}
+
+/**
+ * Get the hash string for the request and refresh functions.
+ */
+function hashFunctions(
+  requestFn: OIDCRequestFunction | null,
+  refreshFn: OIDCRefreshFunction | null
+): string {
+  let requestHash = FN_HASHES.get(requestFn || NO_FUNCTION);
+  let refreshHash = FN_HASHES.get(refreshFn || NO_FUNCTION);
+  if (!requestHash && requestFn) {
+    // Create a new one for the function and put it in the map.
+    FN_HASH_COUNTER++;
```

Comment:
how frequently do we expect to access this codepath and create new functions? if it's a human frequency (e.g., once per second or less) then this approach should be fine - just want to confirm that we don't need to build in a reuse mechanism for the generated numbers

Reply:
It’s at most 2 numbers per MongoClient, that seems fine to me